v2/internal/reconcilers/azure_generic_arm_reconciler.go

```diff
@@ -724,9 +711,13 @@ func (r *AzureDeploymentReconciler) getStatus(ctx context.Context, id string) (g
 
 	// Fill the kube status with the results from the arm status
 	// TODO: The owner parameter here should be optional
-	err = status.PopulateFromARM(knownOwner, reflecthelpers.ValueOfPtr(armStatus)) // TODO: PopulateFromArm expects a value... ick
-	if err != nil {
-		return nil, zeroDuration, errors.Wrapf(err, "converting ARM status to Kubernetes status")
+	if s, ok := status.(genruntime.FromARMConverter); ok {
+		err = s.PopulateFromARM(knownOwner, reflecthelpers.ValueOfPtr(armStatus)) // TODO: PopulateFromArm expects a value... ick
```

Comment:
(I realise you didn't change this but...) Why don't we just change `s.PopulateFromARM` to take a pointer if that would be more convenient? Oh duh, it's code generated, isn't it.

Reply:
It's not that it's code generated (can change that), it's that the call to `json.UnmarshalJSON()` requires the `interface{}` to contain an instance value, not a pointer.

Reply:
I don't get this answer - unmarshalling has already happened at this point, that's what `armStatus` is, right? Unless there's more unmarshalling happening further down?

Reply:
To clarify, I'm talking about changing `s.PopulateFromARM` so that the ARM status is passed in as a pointer (since that's what we have). The method must already have a pointer receiver, since it's updating `s` from the status passed in.

I could see not wanting to do it because passing a pointer means that the ARM status could be modified?